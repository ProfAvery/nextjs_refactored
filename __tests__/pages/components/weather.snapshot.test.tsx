/**
 * @jest-environment jsdom
 */

import { act, render, fireEvent } from "@testing-library/react";
import PageComponentWeather from "../../../pages/components/weather";

describe("PageComponentWeather", () => {
  test("renders correctly", async () => {
    let component: any;
    await act(async () => {
      component =
        await render(<PageComponentWeather></PageComponentWeather>);
    });
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("clicks the h1 element and updates the state", async () => {
    let component: any;
    await act(async () => {
      component = await render(<PageComponentWeather></PageComponentWeather>);
    });
    fireEvent.click(component.container.querySelector("h1"));

    expect(component.asFragment()).toMatchSnapshot();
  });

});
