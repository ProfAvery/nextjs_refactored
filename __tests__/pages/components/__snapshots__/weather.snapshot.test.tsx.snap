// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PageComponentWeather clicks the h1 element and updates the state 1`] = `
<DocumentFragment>
  <h1>
    The weather is sunny, and the counter shows 2
  </h1>
</DocumentFragment>
`;

exports[`PageComponentWeather renders correctly 1`] = `
<DocumentFragment>
  <h1>
    The weather is sunny, and the counter shows 1
  </h1>
</DocumentFragment>
`;
